src/import_whatsapp.js

```diff
@@ -335,9 +334,10 @@ const processText = (text) => {
 		};
 	};
 
-	messages.forEach((message) => {
+	for (const message of messages) {
 		// if the content is valid and there is location or different sender, get the current feature or create a new one and push it to mapdata
 		// we assign it to a variable to be sure the validated content is used
+		// const contribID = await sha256(message.datetime + message.sender); // hash a unique contrib id, this is difficult under more nesting
 
 		if (message.location || message.sender !== currentSender) {
```

Comment:
I think contribID should be generated at the top of this block and passed to createFeature to be added there:

```
if (message.location || message.sender !== currentSender) {
			if (currentFeature && currentFeature.geometry) {
				mapdata.features.push(currentFeature);
			}
                        const contribID = await sha256(message.datetime + message.sender);
			currentFeature = createFeature(message, groupName, contribID);
			currentSender = message.sender;
		}
```

So, it's the datetime of the first message associated with the feature that matters.

Reply:
ah ok
So that's how I originally did it and it was being created for every message regardless of if that message makes it through so I thought I'd try doing it only for the ones that make the cut. Happy to go back to the earlier creation if you'd prefer that way